cfme/tests/services/test_rest_services.py

```diff
@@ -2105,3 +2106,185 @@ def test_deny_service_ordering_via_api(
+                                "type": "DialogFieldDropDownList",
+                                "resource_action": {
+                                    "resource_type": "DialogField",
+                                    "ae_namespace": instance.namespace.name,
+                                    "ae_class": instance.klass.name,
+                                    "ae_instance": instance.name,
+                                },
+                            },
+                        ],
+                    }
+                ],
+            }
+        ],
+    }
+    service_dialog = appliance.rest_api.collections.service_dialogs.action.create(
+        **data
+    )[0]
+    service_catalog = _service_catalogs(request, appliance, num=1)[0]
+    service_template = _service_templates(
+        request,
+        appliance,
+        service_dialog=service_dialog,
+        service_catalog=service_catalog,
+    )[0]
+    yield service_template
+    service_template.action.delete()
+    service_catalog.action.delete()
+    service_dialog.action.delete()
+
+
+@pytest.mark.tier(3)
+@pytest.mark.parametrize("auth_type", ["token", "basic_auth"])
+@test_requirements.rest
+def test_populate_default_dialog_values(
+    appliance, request, auth_type, set_run_automate_method, get_service_template
+):
+    """
+    This test case checks if the default value set for static and dynamic elements are returned
+    when ordering service via API with both basic and token based authentication.
+
+    Polarion:
+        assignee: pvala
+        casecomponent: Rest
+        caseimportance: medium
+        initialEstimate: 1/10h
+        setup:
+            1. Update appliance advanced settings by setting
+                `run_automate_methods_on_service_api_submit` to `true`.
+            2. Create a new domain, namespace, klass, method, schema, and instance.
+                (Method is attached in the Bugzilla 1639413).
+                This method sets value 7 for dynamic element
+            3. Create a Dialog with one static element(name="static") with some default value,
+                and one dynamic element(name="dynamic") with the endpoint pointing to
+                the newly created instance which returns a default value.
+                (value_type="Integer", read_only=True, required=True)
+            4. Create a catalog and create a generic catalog item assigned to it.
+        testSteps:
+            1. Order the catalog item with the given auth_type
+                and check if the response contains default values that were initially set.
+        expectedResults:
+            1. Response must include default values that were initially set.
+
+    Bugzilla:
+        1635673
+        1650252
+        1639413
+        1660237
+    """
+    if auth_type == "token":
+        # obtaining authentication token
+        auth_token_response = requests.get(
+            "https://admin:smartvm@{}/api/auth".format(appliance.hostname), verify=False
```

Comment:
I doubt this has to be hardcoded. please use yaml or appliance values instead

Reply:
Do you mean something like this?
```
credentials = appliance.user.credential
auth_token_response = requests.get(
    appliance.url_path("/api/auth"),
    auth=(credentials.principal, credentials.secret),
    verify=False
)
```